api/lib/application/authentication/oidc/index.js

```diff
@@ -23,6 +24,36 @@ exports.register = async (server) => {
         tags: ['api', 'oidc', 'authentication'],
       },
     },
+    {
+      method: 'POST',
+      path: '/api/oidc/token-reconciliation',
+      config: {
+        auth: false,
+        pre: [
+          {
+            method: featureToggles.checkIfSSOAccountReconciliationIsEnabled,
+            assign: 'checkIfSSOAccountReconciliationIsEnabled',
+          },
+        ],
+        validate: {
+          payload: Joi.object({
+            data: Joi.object({
+              attributes: Joi.object({
+                email: Joi.string().email().required(),
+                password: Joi.string().required(),
+                identity_provider_code: Joi.string().required().valid('POLE_EMPLOI', 'CNAV'),
```

Comment:
```suggestion
                identity_provider: Joi.string().required(),
                authentication_key: Joi.string().required(),
```

- Côté api il n'y a pas de notion de `identity_provider_code`
- Le `lookup` du service registry s'occupe de valider le partenaire
- On se trouve dans une requête POST pour l'authentication_key, c'est plutôt dans le payload que dans les query params vu qu'il n'est pas utilisé pour filtrer le résultat

Reply:
ok pour que le lookup s'occupe de la validation, mais l'erreur retournée est une 500. Il faut la changer pour quelque chose de plus pertinent. Comme je ne compte pas la modifier dans cette PR, je garde la validation Joi et on changera ça au moment opportun

Reply:
Dans ce cas, est-ce qu'il faudrait remettre ce valid sur les autres routes dans cette index ?

Reply:
s'ils ont été enlevés, alors oui :)

Reply:
ils n'avaient juste pas ete mis a la base